examples/reading_events.rs

```diff
@@ -0,0 +1,188 @@
+    ReadStreamOptions,
+};
+use futures::TryStreamExt;
+use serde::{Deserialize, Serialize};
+use std::error::Error;
+use uuid::Uuid;
+
+#[derive(Serialize, Deserialize, Debug)]
+struct TestEvent {
+    pub id: String,
+    pub important_data: String,
+}
+
+type Result<A> = std::result::Result<A, Box<dyn Error>>;
+
+pub async fn read_from_stream(client: &Client) -> Result<()> {
+    // region read-from-stream
+    let result = client
+        .read_stream("some-stream", &Default::default(), All)
+        .await?;
+    // endregion read-from-stream
+
+    // region iterate-stream
+    if let Some(mut events) = result.ok() {
+        while let Some(event) = events.try_next().await? {
+            let test_event = event.get_original_event().as_json::<TestEvent>()?;
+
+            println!("Event> {:?}", test_event);
+        }
+    }
+    // endregion iterate-stream
+
+    Ok(())
+}
+
+pub async fn read_from_stream_position(client: &Client) -> Result<()> {
+    // region read-from-position
+    let options = ReadStreamOptions::default().origin(Origin::Point(10));
+    let result = client.read_stream("some-stream", &options, 20).await?;
+    // endregion read-from-position
+
+    // region iterate-stream
+    if let Some(mut events) = result.ok() {
+        while let Some(event) = events.try_next().await? {
+            let test_event = event.get_original_event().as_json::<TestEvent>()?;
+
+            println!("Event> {:?}", test_event);
+        }
+    }
+    // endregion iterate-stream
+
+    Ok(())
+}
+
+pub async fn read_stream_overriding_user_credentials(client: &Client) -> Result<()> {
+    // region overriding-user-credentials
+    let options = ReadStreamOptions::default()
+        .origin(Origin::Start)
+        .authenticated(Credentials::new("admin", "changeit"));
+
+    let result = client.read_stream("some-stream", &options, All).await?;
+    // endregion overriding-user-credentials
+    Ok(())
+}
+
+pub async fn read_from_stream_position_check(client: &Client) -> Result<()> {
+    // region checking-for-stream-presence
+    let options = ReadStreamOptions::default().origin(Origin::Point(10));
+
+    let result = client.read_stream("some-stream", &options, All).await?;
+
+    match result {
+        ReadResult::Ok(mut events) => {
+            while let Some(event) = events.try_next().await? {
+                let test_event = event.get_original_event().as_json::<TestEvent>()?;
+
+                println!("Event> {:?}", test_event);
+            }
+        }
+
+        ReadResult::StreamNotFound(stream_name) => {
+            println!("Stream not found: {}", stream_name);
+
+            return Ok(());
+        }
+    }
+    // endregion checking-for-stream-presence
+    Ok(())
+}
+
+pub async fn read_stream_backwards(client: &Client) -> Result<()> {
+    // region reading-backwards
+    let options = ReadStreamOptions::default().origin(Origin::End);
+    let result = client.read_stream("some-stream", &options, All).await?;
```

Comment:
**SUGGESTION:** In NodeJS client we decided to move the `maxCount` to options  - as it's optional and JS also doesn't have function overload. See more in https://github.com/EventStore/EventStore-Client-NodeJS/pull/130. Do you think that we could apply that the same way in Rust?

Reply:
Same here instead you use the function that way instead:

```rust
let result = client.read_stream("some-stream", &options, 10).await?;
```

if you want to read `10` events.